model/task_lifecycle.go

```diff
@@ -619,55 +619,35 @@ func UpdateBuildAndVersionStatusForTask(taskId string, updates *StatusChanges) e
 		"task":          t.Id,
 	})
 	loopStart := time.Now()
+	cache := task.NewDisplayTaskCache()
 	// update the build's status based on tasks for this build
 	for _, t := range buildTasks {
 		if !t.IsFinished() {
 			state, _ := t.BlockedState(tasksWithDeps)
 			if state == "blocked" {
-				tasksToNotify += 1
+				tasksToNotify++
 				if evergreen.IsFailedTaskStatus(t.Status) {
 					failedTask = true
 				}
 			}
 			continue
 		}
-		var displayTask *task.Task
-		status := ""
 		finishedTasks++
-		tasksToNotify += 1
+		tasksToNotify++
 
 		startPhaseAt = time.Now()
-		displayTask, err = t.GetDisplayTask()
+		var displayTask *task.Task
+		displayTask, err = cache.Get(&t)
 		if err != nil {
 			return errors.WithStack(err)
 		}
-		if time.Since(startPhaseAt) > slowMS {
```

Comment:
Keep debugging in for now in case there's still a performance problem?

Reply:
It's still in - the things that are out are no longer relevant